tests/cookie_test.cc

```diff
@@ -129,3 +129,25 @@ TEST(cookie_test, invalid_test) {
     ASSERT_THROW(Cookie::fromString("lang=en-US; Max-Age=12ab"), std::invalid_argument);
 }
 
+void addCookies(const char* str, std::function<void (const CookieJar&)> testFunc) {
+    CookieJar jar;
+    jar.addFromRaw(str, strlen(str));
+    testFunc(jar);
+}
+
+TEST(cookie_test, cookiejar_test) {
+    addCookies("key1=value1", [](const CookieJar& jar) {
+        ASSERT_EQ(jar.get("key1").value, "value1");
+    });
+
+    addCookies("key2=value2; key3=value3; key4=; key5=foo=bar", [](const CookieJar& jar) {
+        ASSERT_EQ(jar.get("key2").value, "value2");
+        ASSERT_EQ(jar.get("key3").value, "value3");
+        ASSERT_EQ(jar.get("key4").value, "");
+        ASSERT_EQ(jar.get("key5").value, "foo=bar");
+    });
+    
+    CookieJar jar;
+    ASSERT_THROW(jar.addFromRaw("key4", strlen("key4")), std::runtime_error);
+    ASSERT_THROW(jar.addFromRaw("key4=value4=value4", strlen("key4")), std::runtime_error);
```

Comment:
Why is this expected to throw an exception?  "keys4=value4=value4" should be functionally equivalent to the test that you perform on line #143 with "key5=foo=bar".

I added an extra test to see what would happen.  The test passed, which means that it threw an exception.  I'm not sure why, as the syntax looks correct?

`+    ASSERT_THROW(jar.addFromRaw("key4=value4", strlen("key4")), std::runtime_error);
`

per https://tools.ietf.org/html/rfc6265#section-4.1, "<cookie-value>" can be quoted.
`curl localhost:9080/auth -b 'key1=value1; key2=value2; key3="value3=value4"'
`
That works.  Yeah.

However, I'm undecided on if the Cookie parse should be strict like the RFC states, or follows Postel's law and accepts things that violate the spec, such as:

```
curl localhost:9080/auth -b 'key1="foo,bar baz\p";'
Cookies: [
    key1 = "foo,bar baz\p"
]

Thoughts?
```

Reply:
Ah, sorry, that's totally my mistake. That assert must be an artifact left over when I was messing with test statements. Since the length passed is less than the length of the string passed, it's essentially a duplicate of the previous "ASSERT_THROW".

As for the question of what to accept as a parseable cookie, I'm not sure. If the reversion at https://www.fxsitecompat.com/en-CA/docs/2016/whitespaces-are-no-longer-allowed-in-cookie-names/ is any indication, at least some user application seem to want to use whitespace in their cookies. For what it's worth, the current implementation of "Cookie::fromRaw" appears to take any string as a key or value.